### ISSUE TITLE
Discussion: Restructuring of Modules

### ISSUE THREAD

**Issue by jamesplease** [marionettejs/backbone.marionette#859]
Modules do a lot right now. They can be used for namespacing code and [soon](https://github.com/marionettejs/backbone.marionette/pull/845) as UI/widget components.

There [has been discussion](https://github.com/marionettejs/backbone.marionette/issues/837#issuecomment-32141701) about restructuring modules for version 2 of Marionette.

This issue can be used as a place of discussion for what needs to be done. Some questions:
- [ ] What, exactly, are all of the things modules are used for?  
- [ ] What's a logical way to break them apart?  
- [x] If breaking apart does occur, what should the names of the new objects be?  

// @cobbweb @samccone 

---

**TL;DR**

I'm still unsure why @derickbailey suggested split them into more than 2 things, but the consensus of this conversation thus far has been that at least 2 things make sense. `Modules` for code organization; `Components` for UI widgets.
